### ISSUE TITLE
Make default JAX Zhao mass quadrature accurate for steep integrable cusps

### ISSUE THREAD

**Issue by gomeshun** [gomeshun/jeanspy#54]
Found during the release review of `main` at `c5c94c1a14dde09673907fe10f414966288e247f`. The existing full suite passed (191 tests, 37 subtests), so these reproductions expose coverage gaps.

## Reproduction
```python
import numpy as np
import jax.numpy as jnp
from jeanspy.model_numpyro import ZhaoModel
p = dict(rs_pc=1000., rhos_Msunpc3=.01, r_t_pc=10000., a=1., b=4., g=2.5)
actual = float(ZhaoModel().enclosed_mass(jnp.array(1000.), params=p))
reference = 4*np.pi*p["rhos_Msunpc3"]*p["rs_pc"]**3/(3-p["g"]) * .5**(3-p["g"])
print(actual, reference, actual/reference-1)
# 412629088.0, 177715317.52633464, about +1.322
```
For a=1,b=4 the reference is the analytic mass
`M(r)=4*pi*rho_s*rs**3/(3-g) * (x/(1+x))**(3-g)`.
The mass is integrable for g=2.5. This is a mass-accuracy test, not a claim that every tracer/anisotropy combination admits a physical distribution function.

## Cause
`DMModel.enclosed_mass_numeric` uses 256 uniformly spaced t nodes from 1e-6 to 1. For a steep cusp the integrand scales as t**(2-g); trapezoidal integration overweights the first interval. `auto` selects this path for Zhao to preserve shape-parameter autodiff.

## Scope / acceptance criteria
- Choose and document a supported Zhao parameter/radius envelope.
- Implement a cusp-aware variable transformation/quadrature with finite, trustworthy shape-parameter gradients.
- Expose needed mass-integration controls through the public mass and Jeans solver APIs.
- Validate against analytic a=1,b=4 profiles and independent integration, including near g=3, truncation, float32/64, and JIT/autodiff.
- Add LOS convergence checks so mass errors do not hide behind an outer-grid convergence study.
- If some slopes cannot yet be supported, explicitly document/enforce that restriction instead of silently returning inaccurate masses.

Related: #37 / #44 establish a Plummer+NFW LOS accuracy contract. They explicitly do not guarantee general Zhao accuracy; this issue extends coverage rather than contradicting that contract.
